Updated shaders with new 1.7.8 paths

lbrpdx opened the pull request:
Update after some changes introduced in RetroArch 1.7.8 + took into account feedback I got on Discord about some choices (make handheld shaders more different!). 
Should work on PC x86_64 and RPi3B+. For other platforms and SBC, not sure of the result... and if the GPU is weak, shaders are not a good idea!
Also, when you use GameBoy colorization, be careful of your shader choices - they usually don't mix and match well together.

tcamargo commented:
Default is no shaders if I remember right. Why do you change that?

lbrpdx commented:
Because you still have the option to select « none » rather « auto » (which used to have only a few shaders before).

tcamargo commented:
So, auto means, for most cases, super eagle?

lbrpdx commented:
Super-eagle for CRT-based, lcd3x for LCD/handheld. Anything against super-eagle? Not as pronounced as advanced-aa (=enhanced) and you can still see pixels (but not as much as with the Retro set).

nadenislamarre commented:
hum, not sure that auto setting shaders is a good idea.
auto meaning none seems more logical => no effect.
more over, it will causes issues to several people for who we will have to ask to disable shaders i think.

dmschwartzii commented:
Agreed 

On October 8, 2019 12:41:01 PM CDT, Nicolas Adenis-Lamarre <notifications@github.com> wrote:
>hum, not sure that auto setting shaders is a good idea.
>auto meaning none seems more logical => no effect.
>more over, it will causes issues to several people for who we will have
>to ask to disable shaders i think.
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/batocera-linux/batocera.linux/pull/987#issuecomment-539623902

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

lbrpdx commented:
Ok, we can remove it. But the question is: what is the difference between "auto" and "none" then?

lbrpdx commented:
"auto" used to have GB colorization, by the way, which is probably not a good idea either, if we follow the same logic.

tcamargo commented:
Good question. iIusually set to none. Not sure what auto does...

lbrpdx commented:
How about just removing "auto" then? You would have "none" / "retro" / "scanlines" / "enhanced" / "curvature" and the default set to "none".

nadenislamarre commented:
there is one.
auto means : no user customisation in the batocera.conf file. 
none means : the user won't shaders

at the current time , auto has the same effect as none, but the behavior could change.
for gb colorisations, it is enabled by default because : it seems to have no impact on performances and it is more looking to the real gb.

lbrpdx commented:
I don't think GB colorization should be in the shaders sets: those are two different options from two different ES menus. 
I know this is what we have in the previous iteration (and I can revert to that) but I am not sure it's the best solution.

nadenislamarre commented:
i agree to discuss this point. my first intention was to not have "shaders sets" but more "rendering sets". it means, that shaders are not a part of a renderer. but i'm not sure i had a good idea while it creates confusion.
so the question is more : should we kept shaders together only with nothing else ?
other minds ?

lbrpdx commented:
It would have made sense when all the renderer configuration was through `retroarchcustom.cfg`... but since RetroArch 1.7.8 now separates the shader config with a new CLI `--set-shader XXX` aside from `retroarchcustom.cfg`, I don't think it's relevant any more. 
Shaders should be totally split out and configured on their own now, IMO.

lbrpdx commented:
Any more thought on this?

nadenislamarre commented:
ok to make them separate, but for me, the argument must not be "because it is like this in retroarch", we must think userfriendly first, then, we model as we want.
in this pr, auto is not on no shader. we're not aligned on that point ?

lbrpdx commented:
Auto is now on "no shader", and I think we should remove "auto" from the ES shader menu options. Now, basically, auto=none. I can do the corresponding PR in `batocera-emulationstation` if you're OK with it.

lbrpdx commented:
Alright, since you want to keep an "auto" option and shut down the ES change, I believe you can merge this one.

lbrpdx commented:
I'll update the wiki to explain that auto = none for the moment, but can change in the future.

nadenislamarre commented:
why having removed gba colorization by default and scanline=false by default ?

lbrpdx commented:
Because you said, wrt colorization and shaders:
> "ok to make them separate"

So I removed the colorization from the shader config.

And I removed `scanlines=false` because I thought we wanted to have no shader option at all set by default for "auto"... so just let whatever the emulators implement by default. You can put it back, not sure how impactful it is really.

nadenislamarre commented:
in fact, you cannot just change that withtout altering some other file, because it will break the feature as default. we can do an other pr for that separation.
just reset the file package/batocera/emulators/retroarch/shaders/batocera-shaders/configs/rendering-defaults.yml  and this pr is ok for validation.
